pkg/resources/api_authentication_integration_common.go

```diff
@@ -0,0 +1,376 @@
+		}
+	}
+	oauthClientId, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_CLIENT_ID" })
+	if err == nil {
+		if err = d.Set("oauth_client_id", oauthClientId.Value); err != nil {
+			return err
+		}
+	}
+	oauthClientAuthMethod, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+		return property.Name == "OAUTH_CLIENT_AUTH_METHOD"
+	})
+	if err == nil {
+		if err = d.Set("oauth_client_auth_method", oauthClientAuthMethod.Value); err != nil {
+			return err
+		}
+	}
+	oauthTokenEndpoint, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_TOKEN_ENDPOINT" })
+	if err == nil {
+		if err = d.Set("oauth_token_endpoint", oauthTokenEndpoint.Value); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func handleApiAuthRead(d *schema.ResourceData,
+	integration *sdk.SecurityIntegration,
+	properties []sdk.SecurityIntegrationProperty,
+	withExternalChangesMarking bool,
+	extraFieldsDescribeMappings []describeMapping,
+) error {
+	if err := d.Set("name", integration.Name); err != nil {
+		return err
+	}
+	if err := d.Set("comment", integration.Comment); err != nil {
+		return err
+	}
+	if err := d.Set("enabled", integration.Enabled); err != nil {
+		return err
+	}
+	if withExternalChangesMarking {
+		oauthAccessTokenValidity, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "OAUTH_ACCESS_TOKEN_VALIDITY"
+		})
+		if err != nil {
+			return err
+		}
+
+		oauthRefreshTokenValidity, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "OAUTH_REFRESH_TOKEN_VALIDITY"
+		})
+		if err != nil {
+			return err
+		}
+
+		oauthClientId, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_CLIENT_ID" })
+		if err != nil {
+			return err
+		}
+
+		oauthClientAuthMethod, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "OAUTH_CLIENT_AUTH_METHOD"
+		})
+		if err != nil {
+			return err
+		}
+
+		oauthTokenEndpoint, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_TOKEN_ENDPOINT" })
+		if err != nil {
+			return err
+		}
+
+		oauthAccessTokenValidityInt, err := strconv.Atoi(oauthAccessTokenValidity.Value)
+		if err != nil {
+			return err
+		}
+		oauthRefreshTokenValidityInt, err := strconv.Atoi(oauthRefreshTokenValidity.Value)
+		if err != nil {
+			return err
+		}
+
+		if err = handleExternalChangesToObjectInDescribe(d,
+			append(extraFieldsDescribeMappings,
+				describeMapping{"oauth_access_token_validity", "oauth_access_token_validity", oauthAccessTokenValidityInt, oauthAccessTokenValidityInt, stringToIntNormalizer},
```

Comment:
may be wrong, but those normalizers could be removed (?)

Reply:
I think without them comparisons failed because of different types.